src/validations/user.schema.ts

```diff
@@ -0,0 +1,28 @@
+import { z } from "zod";
+
+export const userSchema = z.object({
+  name: z.string().optional(),
+  username: z
+    .string()
+    .min(3, "Username must be at least 3 characters")
+    .max(20, "Username must be at most 20 characters")
+    .regex(/^[a-zA-Z0-9][a-zA-Z0-9._]*$/, "Username must start with a letter or number and contain only letters, numbers, underscores, or dots"),
+  email: z.string().email("Invalid email format"),
+  birthDate: z
+    .string()
+    .refine(value => {
+      const date = new Date(value);
+      return !isNaN(date.getTime()) && date >= new Date("1904-01-01") && date <= new Date(new Date().setFullYear(new Date().getFullYear() - 13));
+    }, { message: "Invalid date format or out of range" })
+    .transform(value => new Date(value)),

```

Comment:
acredito que essa verificação deveria ser dinâmica com base na data atual (`between 13 and 120 years old`) e não uma data fixa (ex: `1904-01-01`). 

além disso, ao invés de jogar em um `.refine()`, seria possível utilizar os helpers `.date()`, `.min()` e `.max()` pra fazer essa verificação de forma mais sucinta